index.js

```diff
@@ -37,9 +38,18 @@ function compose (middleware) {
     function dispatch (i) {
       if (i <= index) return Promise.reject(new Error('next() called multiple times'))
       index = i
-      const fn = middleware[i] || next
-      if (!fn) return Promise.resolve()
+      let fn = middleware[i]
       try {
+        if (fn && context.wrappers) {
+          context.wrappers.forEach((wrapper) => { fn = wrapper(fn) })
+        }
+        fn = fn || next
+        if (!fn) return Promise.resolve()
+        if (typeof fn !== 'function') throw new TypeError('Middleware must be a function!')
+        if (isGeneratorFunction(fn)) {
+          throw new TypeError('Middleware should not be a generator! ' +
+            'Are you sure you didn\'t forget a wrapper?')
+        }
```

Comment:
I just made these changes in a seperate commit, but the same PR. I think it makes sense to move these checks to koa-convert now that we have wrappers.